backend/graminsta/core/services.py

```diff
@@ -33,3 +35,34 @@ def create_userinfo(validated_data):
         gender=validated_data.pop('gender')
     )
     return user_info
+
+
+def user_authentication(request):
+    """User Authentication
+
+    Parameters
+    ----------
+    request: rest_framework.request.Request
+        Request from the frontend
+
+    Returns
+    -------
+    token.key: String format
+        if the authentication passed.
+    None if the authentication failed.
+    """
+    data = request.data
+    user = authenticate(
+        request,
+        username=data.get("username", ""),
+        password=data.get("password", ""))
+    if user is not None:
+        login(request, user)
```

Comment:
I think `login` is to add some instructions to let client to set cookie for authentication, do we need that?